electrum/wallet.py

```diff
@@ -1952,6 +1972,14 @@ def fee_estimator(size: Union[int, float, Decimal]) -> int:
                     amount = change[0].value
                     if amount <= self.lnworker.num_sats_can_receive():
                         tx.replace_output_address(change[0].address, DummyAddress.SWAP)
+
+            self.logger.debug(f'coinchooser returned tx with {len(tx.inputs())} inputs and {len(tx.outputs())} outputs')
+            to_be_consumed = sum(c.value_sats() for c in tx.inputs() if c in coins)
```

Comment:
why is this only looking at `coins`? what about `inputs`?

Reply:
the parameter `inputs` is used for sweeping coins that are not in the wallet. These inputs should not be added to `to_be_consumed`, because this would overestimate how much the transaction is spending. 

We could add an assert that inputs are not wallet coins.